include/boost/unordered/detail/foa.hpp

```diff
@@ -1129,6 +1129,25 @@ _STL_RESTORE_DEPRECATED_WARNING
  */
 constexpr static float const mlf = 0.875f;
 
+template <class T>
+union storage
+{
+  T t_;
+  storage(){}
+  ~storage(){}
+};
+
+template <class TypePolicy,class A,class T>
+struct drop_guard
```

Comment:
I know I asked for this to be defined outside `table`, but upon re-reviewing I think it makes more sense to rename this as `destroy_on_exit` and define it in-class alongside `clear_on_exit` and `erase_on_exit`.